### DISCUSSION TITLE
Wallet pop up is overeager on requesting eth_accounts

### DISCUSSION THREAD

**Discussion by wengzilla** [phantom/docs#103] [Help & Support] [answered]
Latest version of Phantom wallet is overeager to open. It's popping open anytime I request 'eth_accounts'. AFAIK, this is non-standard behavior and is highly disruptive. Metamask does not pop open.

```
const accounts = await window.ethereum.request({
   method: 'eth_accounts'
});
```

**Comment by adamdelphantom** [comment]
@wengzilla Do you have a hosted application where I can reproduce this? Also, what does the wallet render when it pops? A screenshot would be helpful. Thanks!

**Reply by wengzilla** [reply]
Sure. Here's a simple example: https://phantom-bug.vercel.app/

Code is as follows:

```html
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <meta http-equiv="X-UA-Compatible" content="ie=edge" />
    <meta name="description" content="baru belajar html" />
  </head>
  <body>
    <h1>Phantom Test</h1>
    <button id="button">Click Me</button>
    <div>Accounts Connected:</div>
    <div id="account-list"></div>

    <script type="text/javascript">
      async function fetchAccounts() {
        try {
          let accounts = await window.ethereum.request({
            method: 'eth_accounts',
          });
          document.getElementById('account-list').innerHTML = accounts.join(', '); // Display accounts as a comma-separated list
          console.log(accounts);
        } catch (error) {
          console.error(error);
        }
      }

      // Add an event listener to the button element
      document.getElementById('button').addEventListener('click', fetchAccounts);
    </script>
  </body>
</html>
```

**Reply by wengzilla** [reply]
@adamdelphantom are you able to repro?

**Reply by adamdelphantom** [reply]
No, when I navigate to your app the wallet only pops once to connect and no other times. What browser/OS are you using?

**Reply by wengzilla** [reply]
Just repeating here... Fetch accounts should _not_ trigger Phantom to pop up. When you click the button, Metamask does not open but Phantom does. So if you see the pop, you're reproducing the issue already. Does that make sense?

**Reply by adamdelphantom** [reply]
Hey @wengzilla, yes that makes sense. If there is no active connection what are you expecting to happen when you query for accounts?

**Reply by wengzilla** [reply]
So I think the behavior should replicate that of Metamask. If wallet is connected, it should return the array of connected addresses (e.g., `[0x123.., 0x456...]`. If the wallet is not connected, it should return an empty array (e.g., `[]`). In either case, popping up the connect wallet modal should not happen.

**Reply by wengzilla** [reply]
@adamdelphantom just checking... Is this being fixed? This issue is still causing Phantom to pop on my website. Will have to continue actively discouraging my users from using Phantom while this is ongoing.

**Reply by adamdelphantom** [reply]
@wengzilla Thanks for your patience! This happens when the Phantom Default App Wallet setting is set to "Always Ask." Which is the default setting. In order to avoid this on the short term your users can set the preference to either Metamask or Phantom in Settings > Preferences > Default App Wallet.

We also support https://eip6963.org/ which will eventually remove the need for the Always Ask setting.

**Reply by wengzilla** [reply]
Yes, I understand that it pops when the setting is "Always Ask", but I'm not convinced that popping is the right behavior when a site is merely asking for the connected `eth_accounts`. As far as I can tell, Metamask does not pop open, so it's unclear why Phantom should.

Do you believe aggressively popping is the correct behavior? If so, does that mean you will not fix this issue?

**Reply by adamdelphantom** [reply]
My understanding is that with as support for eip3963 become pervasive, we will update this behavior. For the time being the responsiveness is necessary for Phantom users to be able to use Phantom with dApps that don't explicitly list Phantom as an option.

**Reply by wengzilla** [reply]
Can you link me to EIP-3963? I'm unable to find it and am not familiar with that proposal. Thanks!

Also, I do have other EVM-focused wallets and none of them seem to aggressively pop open on request `eth_account`. Do you know of any others with this type of behavior?

**Reply by adamdelphantom** [reply, marked as answer]
Apologies, I mistyped the proposal title. This is the proposal https://eips.ethereum.org/EIPS/eip-6963 and a reference implementation https://eip6963.org.

I haven't personally tested other wallets for this functionality.

**Comment by wengzilla** [comment]
It shouldn't pop at all... if you disable Phantom, you will recognize that
Metamask doesn't pop on the same method call.
